### PR TITLE
sigup endpoint with hash password

### PR DESCRIPTION
**Pull request by muhammadharoun**

### CODE REVIEW

**Review by rawand1998 — APPROVED** [review]

**Review by NourAdawi — APPROVED** [review]

File: server/database/queries/addUserQuery.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const pool = require('../connection');
+
+const addUserQuery = (name, email, password, isAdmin, imgUrl) => pool.query(`INSERT INTO users (name, email, password, isAdmin, imgUrl) VALUES
+('${name}', '${email}', '${password}', ${isAdmin}, '${imgUrl}');`);
```
Comment (Mu7ammadAbed) [inline server/database/queries/addUserQuery.js:4]:
```suggestion
const connection = require('../connection');

const addUserQuery = (name, email, password, isAdmin, imgUrl) => connection.query('INSERT INTO users (name, email, password, isAdmin, imgUrl) VALUES
($1, $2, $3, $4, $5);', [name, email, password, isAdmin, imgUrl]);
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/database/connection.js [hunk 2]
```diff
@@ -2,7 +2,6 @@ require('env2')('.env');
 const { Pool } = require('pg');
 
 const { DB_URL } = process.env;
```
Comment (Mu7ammadAbed) [inline server/database/connection.js:4]: You should add here a switch case for the NODE_ENV

File: server/routes/addUser.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const { addUserQuery } = require('../database/queries');
+const { hashPassword } = require('../middlewares/hashPassword');
+
+const addUserHandler = (req, res) => {
+  const {
+    name, email, password, isAdmin, imgUrl,
+  } = req.body;
```
Comment (Mu7ammadAbed) [inline server/routes/addUser.js:7]: There's no validation for these inputs?

File: server/routes/addUser.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  hashPassword(password)
+    .then((hashpass) => {
+      addUserQuery(name, email, hashpass, isAdmin, imgUrl)
+        .then(() => res.json({ status: 'scusses' }))
+        .catch((err) => res.json({ status: 'err', error: err }));
+    })
+    .catch((err) => res.json({ status: 'err', error: err }));
```
Comment (Mu7ammadAbed) [inline server/routes/addUser.js:14]: status here should appoint to the statusCode

File: server/routes/addUser.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+const { hashPassword } = require('../middlewares/hashPassword');
+
+const addUserHandler = (req, res) => {
+  const {
+    name, email, password, isAdmin, imgUrl,
+  } = req.body;
+  hashPassword(password)
+    .then((hashpass) => {
+      addUserQuery(name, email, hashpass, isAdmin, imgUrl)
+        .then(() => res.json({ status: 'scusses' }))
```
Comment (Mu7ammadAbed) [inline server/routes/addUser.js:11]: Did you check for the cases where the email exists for example?

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Pull request merged**